From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 8 Aug 2017 03:49:53 +0200
Subject: [PATCH] network: public, mgmt: Use arm-pod3 POD config

FIXME: This overwrites virtual POD config, breaking virtual deploys.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
fi
---
 mcp/config/states/networks                         |  2 +-
 .../baremetal-mcp-ocata-ovs-ha/infra/init.yml      | 20 ++++----
 .../baremetal-mcp-ocata-ovs-ha/openstack/init.yml  | 56 +++++++++++-----------
 mcp/salt-formulas/armband/external_gateway.sls     | 12 +++++
 mcp/scripts/net_public.xml                         |  2 +-
 mcp/scripts/salt.sh                                |  2 +
 6 files changed, 54 insertions(+), 40 deletions(-)
 create mode 100644 mcp/salt-formulas/armband/external_gateway.sls

diff --git a/mcp/config/states/networks b/mcp/config/states/networks
index df4c0bb..230fd44 100755
--- a/mcp/config/states/networks
+++ b/mcp/config/states/networks
@@ -1,3 +1,3 @@
 salt 'ctl01*' cmd.run ". /root/keystonercv3; openstack compute service list; openstack network agent list; openstack stack list; openstack volume service list"
 salt 'ctl01*' cmd.run ". /root/keystonercv3; openstack network create --external --default --provider-network-type flat --provider-physical-network physnet1 floating_net"
-salt 'ctl01*' cmd.run ". /root/keystonercv3; openstack subnet create --gateway 10.0.9.254 --no-dhcp --allocation-pool start=10.0.9.130,end=10.0.9.200 --network floating_net --subnet-range 10.0.9.0/24 floating_subnet"
+salt 'ctl01*' cmd.run ". /root/keystonercv3; openstack subnet create --gateway 10.0.8.254 --no-dhcp --allocation-pool start=10.0.8.130,end=10.0.8.200 --network floating_net --subnet-range 10.0.8.0/24 floating_subnet"
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
index f3e12f4..b09d1cf 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
@@ -16,20 +16,20 @@ parameters:
     infra_maas_database_password: opnfv_secret

     # infra service addresses
-    infra_config_address: 10.167.4.100
+    infra_config_address: 172.16.10.100
     infra_config_deploy_address: 192.168.10.100
-    infra_maas_node01_address: 10.167.4.250
+    infra_maas_node01_address: 172.16.10.250
     infra_maas_node01_deploy_address: 192.168.10.250
     infra_maas_node01_pxe_address: 192.168.11.250
-    infra_maas_node01_external_address: 10.16.0.250
-    infra_compute_node01_address: 10.167.4.141
-    infra_compute_node02_address: 10.167.4.142
-    infra_compute_node03_address: 10.167.4.143
+    infra_maas_node01_external_address: 10.0.8.250
+    infra_compute_node01_address: 172.16.10.141
+    infra_compute_node02_address: 172.16.10.142
+    infra_compute_node03_address: 172.16.10.143

-    infra_kvm_address: 10.167.4.140
-    infra_kvm_node01_address: 10.167.4.141
-    infra_kvm_node02_address: 10.167.4.142
-    infra_kvm_node03_address: 10.167.4.143
+    infra_kvm_address: 172.16.10.140
+    infra_kvm_node01_address: 172.16.10.141
+    infra_kvm_node02_address: 172.16.10.142
+    infra_kvm_node03_address: 172.16.10.143

     infra_maas_node01_hostname: mas01
     infra_kvm_node01_hostname: kvm01
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
index 4036be1..dce7fb7 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
@@ -3,9 +3,9 @@ parameters:

     openstack_version: ocata

-    openstack_gateway_node01_address: 10.167.4.124
-    openstack_gateway_node02_address: 10.167.4.125
-    openstack_gateway_node03_address: 10.167.4.126
+    openstack_gateway_node01_address: 172.16.10.124
+    openstack_gateway_node02_address: 172.16.10.125
+    openstack_gateway_node03_address: 172.16.10.126
     openstack_gateway_node01_tenant_address: 10.1.0.6
     openstack_gateway_node02_tenant_address: 10.1.0.7
     openstack_gateway_node03_tenant_address: 10.1.0.9
@@ -14,21 +14,21 @@ parameters:
     openstack_gateway_node03_hostname: gtw03

     # openstack service addresses
-    openstack_proxy_address: 10.167.4.80
-    openstack_proxy_node01_address: 10.167.4.81
-    openstack_proxy_node02_address: 10.167.4.82
-    openstack_control_address: 10.167.4.10
-    openstack_control_node01_address: 10.167.4.11
-    openstack_control_node02_address: 10.167.4.12
-    openstack_control_node03_address: 10.167.4.13
-    openstack_database_address: 10.167.4.50
-    openstack_database_node01_address: 10.167.4.51
-    openstack_database_node02_address: 10.167.4.52
-    openstack_database_node03_address: 10.167.4.53
-    openstack_message_queue_address: 10.167.4.40
-    openstack_message_queue_node01_address: 10.167.4.41
-    openstack_message_queue_node02_address: 10.167.4.42
-    openstack_message_queue_node03_address: 10.167.4.43
+    openstack_proxy_address: 172.16.10.80
+    openstack_proxy_node01_address: 172.16.10.81
+    openstack_proxy_node02_address: 172.16.10.82
+    openstack_control_address: 172.16.10.10
+    openstack_control_node01_address: 172.16.10.11
+    openstack_control_node02_address: 172.16.10.12
+    openstack_control_node03_address: 172.16.10.13
+    openstack_database_address: 172.16.10.50
+    openstack_database_node01_address: 172.16.10.51
+    openstack_database_node02_address: 172.16.10.52
+    openstack_database_node03_address: 172.16.10.53
+    openstack_message_queue_address: 172.16.10.40
+    openstack_message_queue_node01_address: 172.16.10.41
+    openstack_message_queue_node02_address: 172.16.10.42
+    openstack_message_queue_node03_address: 172.16.10.43


     openstack_telemetry_hostname: mdb
@@ -36,18 +36,18 @@ parameters:
     openstack_telemetry_node02_hostname: mdb02
     openstack_telemetry_node03_hostname: mdb03

-    openstack_telemetry_address: 10.167.4.75
-    openstack_telemetry_node01_address: 10.167.4.76
-    openstack_telemetry_node02_address: 10.167.4.77
-    openstack_telemetry_node03_address: 10.167.4.78
+    openstack_telemetry_address: 172.16.10.75
+    openstack_telemetry_node01_address: 172.16.10.76
+    openstack_telemetry_node02_address: 172.16.10.77
+    openstack_telemetry_node03_address: 172.16.10.78

     # OpenStack Compute
-    openstack_compute_node01_single_address: 10.167.4.101
-    openstack_compute_node02_single_address: 10.167.4.102
-    openstack_compute_node03_single_address: 10.167.4.103
-    openstack_compute_node01_control_address: 10.167.4.101
-    openstack_compute_node02_control_address: 10.167.4.102
-    openstack_compute_node03_control_address: 10.167.4.103
+    openstack_compute_node01_single_address: 172.16.10.101
+    openstack_compute_node02_single_address: 172.16.10.102
+    openstack_compute_node03_single_address: 172.16.10.103
+    openstack_compute_node01_control_address: 172.16.10.101
+    openstack_compute_node02_control_address: 172.16.10.102
+    openstack_compute_node03_control_address: 172.16.10.103
     openstack_compute_node01_tenant_address: 10.1.0.101
     openstack_compute_node02_tenant_address: 10.1.0.102
     openstack_compute_node03_tenant_address: 10.1.0.103
diff --git a/mcp/salt-formulas/armband/external_gateway.sls b/mcp/salt-formulas/armband/external_gateway.sls
new file mode 100644
index 0000000..d8760fe
--- /dev/null
+++ b/mcp/salt-formulas/armband/external_gateway.sls
@@ -0,0 +1,12 @@
+set_routes:
+  network.routes:
+    # TODO: external_interface needs a better name
+    # - name: {{ salt['pillar.get']('_param:external_interface') }}
+    - name: enp4s0
+    - routes:
+      - name: external_gw
+        ipaddr: 0.0.0.0
+        netmask: 0.0.0.0
+        # TODO: define gateway IP as parameter
+        # gateway: {{ salt['pillar.get']('_param:external_gateway') }}
+        gateway: 10.0.8.254
diff --git a/mcp/scripts/net_public.xml b/mcp/scripts/net_public.xml
index f455480..5ecb658 100644
--- a/mcp/scripts/net_public.xml
+++ b/mcp/scripts/net_public.xml
@@ -2,5 +2,5 @@
   <name>public</name>
   <bridge name="public"/>
   <forward mode="nat"/>
-  <ip address="10.0.9.254" netmask="255.255.255.0" />
+  <ip address="10.0.8.254" netmask="255.255.255.0" />
 </network>
diff --git a/mcp/scripts/salt.sh b/mcp/scripts/salt.sh
index 5732f4c..5ec6708 100755
--- a/mcp/scripts/salt.sh
+++ b/mcp/scripts/salt.sh
@@ -64,6 +64,8 @@ ssh ${SSH_OPTS} "${SSH_SALT}" bash -s << SALT_INSTALL_END

   salt -C 'I@salt:master' state.sls linux
   salt -C '* and not cfg01*' state.sls linux
+  # TODO: probably requires network restart, does not work yet
+  # salt -C '* and not cfg01* and not cmp*' state.sls armband.external_gateway

   salt '*' state.sls ntp
 SALT_INSTALL_END
